### ISSUE TITLE
The process of video detection and counting is very low

### ISSUE THREAD

**Issue by ZYT1122** [nicholaskajoh/Vehicle-Counting#28]
hi, thanks for your job , when I use this demo ,  The process of video detection and counting is very low！  FPS is very low。。。  why？  And  I found it nothing to do with gpu。。。。My gpu is 2080TI。  I tried to reduce the detection  interest category, but it can't useful 
So could you help me  solve this problem？  Thank you very much

**Comment by ZYT1122** [comment]
I have another very urgent question！   
When I run this car detection count demo, I found that there is a problem with the object boxes. The object boxes is gone when the counting line has not been reached.  Why is this?  As shown below:
Notice the second yellow car in the picture！！

**Comment by ZYT1122** [comment]
![1](https://user-images.githubusercontent.com/43988635/63918853-1e006100-ca70-11e9-8868-46260962318c.png)
![2](https://user-images.githubusercontent.com/43988635/63918857-1fca2480-ca70-11e9-90b4-990d5e5cc4f8.png)
![3](https://user-images.githubusercontent.com/43988635/63918859-2193e800-ca70-11e9-8cd1-89e41731dfa5.png)

**Comment by nicholaskajoh** [comment]
The detector currently runs on CPU. Support for GPU will be added soon. Also, updates to the vehicle trackers run sequentially. It'll be updated to run in parallel. These will dramatically speed up the system. Stay tuned! :)
The issue of disappearing blobs/bounding boxes is caused by detection and tracking failures. You can use these parameters (`mcdf ` and `mctf `) to tune the system in order to get better perf. See the [config section of the README](https://github.com/nicholaskajoh/Vehicle-Counting#configuration) for more info.

**Comment by ZYT1122** [comment]
Thank you very much for your reply.！I will definitely pay attention to the GPU version! On the "mcdf and mctf" two parameters I saw "config section of the README" here but I still do not understand the meaning of these two parameters, can you please help me explain these two parameters in detail? Thank you very much! ! !

**Comment by nicholaskajoh** [comment]
The detector(s) and tracker(s) fail to ID/track objects every now and then. If we're unable to detect/track a vehicle, we assume it has left the frame. Due to false negatives, however, we can't be sure that an object has actually left the frame if a tracker or detector says so. We want at least _n_ consecutive detection and tracking failures (which you can set) before we conclude that a vehicle has actually left the frame in order to accommodate these failures.

**Comment by nicholaskajoh** [comment]
GPU support has now been added for the YOLO detector (see #34, TFODA in the works). From my tests, processing speed has more than doubled! There's still much to be desired in terms of the performance of the VCS though. I did some work on using multiprocessing in Python to increase tracking speed but it didn't bear any fruit because the OpenCV trackers (C++ code) couldn't be pickled. I'm still looking for ways to optimize this part of the code and I'm open to suggestions and contributions. Cheers!
